pkg/build/controller/strategy/util.go

```diff
@@ -90,3 +65,49 @@ func setupBuildEnv(build *buildapi.Build, pod *kapi.Pod) error {
 	}
 	return nil
 }
+
+// setupDockerSecrets mounts Docker Registry secrets into Pod running the build,
+// allowing Docker to authenticate against private registries or Docker Hub.
+func setupDockerSecrets(pod *kapi.Pod, secretRef *kapi.ObjectReference) {
+	if secretRef == nil {
+		return
+	}
+
+	// TODO: Should namespace be required here?
+	namespace := secretRef.Namespace
+	if len(namespace) == 0 {
+		namespace = pod.Namespace
+	}
+
+	if len(namespace) == 0 {
+		glog.Infof("Unable to get namespace from RegistrySecret, using default")
+		namespace = kapi.NamespaceDefault
```

Comment:
Right, it should inherit name space from Pod, which inherits it from
BuildConfig
On Mar 23, 2015 7:40 PM, "Jordan Liggitt" <notifications@github.com> wrote:

> In pkg/build/controller/strategy/util.go
> <https://github.com/openshift/origin/pull/1411#discussion_r26965974>:
>
> > +// setupDockerSecrets mounts Docker Registry secrets into Pod running the build,
> > +// allowing Docker to authenticate against private registries or Docker Hub.
> > +func setupDockerSecrets(pod *kapi.Pod, secretRef *kapi.ObjectReference) {
> > +	if secretRef == nil {
> > +		return
> > +	}
> > +
> > +	// TODO: Should namespace be required here?
> > +	namespace := secretRef.Namespace
> > +	if len(namespace) == 0 {
> > +		namespace = pod.Namespace
> > +	}
> > +
> > +	if len(namespace) == 0 {
> > +		glog.Infof("Unable to get namespace from RegistrySecret, using default")
> > +		namespace = kapi.NamespaceDefault
>
> not sure we're even going to allow using secrets outside your namespace
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/1411/files#r26965974>.
>